onnxruntime/core/framework/finalize_session_state.cc

```diff
@@ -181,6 +183,15 @@ common::Status SaveInitializedTensors(const Env& env, const std::basic_string<PA
     ORT_RETURN_IF_ERROR(ort_value_name_idx_map.GetIdx(entry.first, ort_value_index));
     id_to_initialized_tensor[ort_value_index] = entry.second;
   }
+  
+  // tensors requiring a specific allocation order are traced first, to ensure they are allocated in order
+  auto initialized_tensors_to_allocate = id_to_initialized_tensor;
+  for (int ort_value_index : initializer_allocation_order) {
```

Comment:
>initializer_allocation_order [](start = 29, length = 28)

Does this need to be per-device? Some models may have weights on CPU, i.e. embedding.

Reply:
I believe execution plan is device agnostic, but memory pattern generation is device specific, here we trace the ort_value_indices which is part of execution plan. I can double check.

---
In reply to: [453877764](https://github.com/microsoft/onnxruntime/pull/4443#discussion_r453877764) [](ancestors = 453877764)